### PR TITLE
Load Balancers doc explaining proxy protocol labels

### PR DESCRIPTION
**Pull request by mattmanning**

### CODE REVIEW

File: _deployment/load-balancers.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: "Load Balancers"
+order: 950
+---
+
+Convox provisions a load balancer for each process type in your app that exposes external ports. The load balancer provides a static domain and distributes incoming requests to your processes.
+
+### Default TCP / SSL
+
+By default the load balancer speaks TCP. When you add a certificate as described in the [SSL document](/docs/ssl), the balancer can accept SSL-encrypted requests. It decrypts these requests and passes them to your process via plain TCP.
```
Comment (ddollar) [inline _deployment/load-balancers.md]: This section seems to know that the reader knows / cares about the difference between TCP and HTTP routing. Can probably just replace this entire section with a description of how the load balancer works in terms like "can load balance any arbitrary TCP point". Add an info panel to talk about standard proxy headers and why they are missing.

File: _deployment/load-balancers.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+### Proxy Protocol
+
+When the balancer is speaking TCP it does not inject any HTTP headers into requests. Many apps expect headers such as `X-Forwarded-Proto` (to determine whether the request is secure) and `X-Forwarded-For` (to determine the originating IP address of the client).
```
Comment (ddollar) [inline _deployment/load-balancers.md]: This section can probably be worked into the info notice above. Proxy protocol by itself does not fix the HTTP header problem or really anything except provide the real source IP address. Can drop all of the technical description of how it works and simply describe how to turn it on. Also talk about something along the lines of "Your app will need to receive and handle this protocol. See App Termination below."

Comment (ddollar) [inline _deployment/load-balancers.md]: Can delete the rest of this section. After that let's add a section about why you might want to set up Haproxy or Nginx in your app. (ELB <-> Container encryption, proxy protocol handling, SSL header injection, etc)

**Comment by nzoschke** [comment]
Great to see some docs around this. I see a trend of keeping docs less technical, and more real-world use case driven. Some ideas how to apply that to this doc...

First section is `TCP`, and explains that we use this because it works for websockets, databases and more.

Next section is `SSL`, and explains that in general you want to do HTTPS / TLS everywhere, 

Final section is `Client IPs` and explains how to set up an app to get `X-Forwarded-For` if you want or need that.

Might not even say "Proxy Protocol" if possible...

**Comment by nzoschke** [comment]
Or yeah, final section is `App Level Enhancements' and a recipe for doing HAProxy to get X-Forwarded-For, SSL -> SSL, and intra-container load balancing for a unicorn style server.

File: _deployment/load-balancers.md (hunk 2 above)
Comment (ddollar) [inline _deployment/load-balancers.md]: Overall structure of this doc can be:

# Load Balancing
basic information on what to expect, probably a link to port mappings here

### Advanced Configuration

#### End-to-end Encryption
talk about haproxy/nginx and self-signed cert

#### Header Injection
talk about SSL headers
(panel on proxy protocol)

**Comment by ddollar** [comment]
Closed in favor of #125 with the new labels upadtes

**Pull request closed without merging**